### PR TITLE
test: Add Windows subprocess impersonation tests

### PR DESCRIPTION
**Pull request by roblutt**
### What was the problem/requirement? (What/Why)
We want the subprocess module to be unit tested for impersonation on Windows to ensure code quality.

### What was the solution? (How)
Add unit tests.

### What is the impact of this change?
All tests run locally. All tests except `test_notify_ends_process` run in Github Actions.

We'll investigate the `test_notify_ends_process` failure once #58 is merged.

****

### How was this change tested?
Unit tests pass.

### Was this change documented?
No

### Is this a breaking change?
No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

**Review by ddneilson — CHANGES_REQUESTED** [review]

**Review by ddneilson — DISMISSED** [review]

**Review by ddneilson — DISMISSED** [review]

File: test/openjd/sessions/test_subprocess.py [hunk 1]
```diff
@@ -622,3 +633,222 @@ def end_proc():
...
+        windows_user: WindowsSessionUser,
+    ) -> None:
+        # Make sure that process is sent a notification signal
+
+        # GIVEN
+        logger = build_logger(queue_handler)
+        python_app_loc = (Path(__file__).parent / "support_files" / "app_20s_run.py").resolve()
+        subproc = LoggingSubprocess(
+            logger=logger,
+            args=["powershell", "-command", f"python {str(python_app_loc)}"],
```
Comment (matta-aws) [inline test/openjd/sessions/test_subprocess.py]: We don't need `powershell` here.  We can call python directly.  ([as in same user version](https://github.com/OpenJobDescription/openjd-sessions-for-python/pull/72))

File: test/openjd/sessions/test_subprocess.py [hunk 2]
```diff
@@ -622,3 +633,222 @@ def end_proc():
...
+    ) -> None:
+        # Make sure that the subprocess is forcefully killed when terminated
+
+        # GIVEN
+        logger = build_logger(queue_handler)
+        python_app_loc = (Path(__file__).parent / "support_files" / "app_20s_run.py").resolve()
+
+        subproc = LoggingSubprocess(
+            logger=logger,
+            args=["powershell", "-Command", "python", str(python_app_loc)],
```
Comment (matta-aws) [inline test/openjd/sessions/test_subprocess.py]: Same here.  Can just call python/sys.executable

File: test/openjd/sessions/test_subprocess.py [hunk 3]
```diff
@@ -622,3 +633,222 @@ def end_proc():
...
+            wait((future1, future2), return_when="ALL_COMPLETED")
+
+        # THEN
+        assert not subproc.is_running
+        messages = collect_queue_messages(message_queue)
+        # If we printed "Trapped" then we hit our signal handler, and that shouldn't happen.
+        assert "Trapped" not in messages
+        # Check for the first message that would print
+        assert "Log from test 0" in messages
+        # If there's no 9, then we ended before the app naturally finished.
```
Comment (matta-aws) [inline test/openjd/sessions/test_subprocess.py]: nit:  19 (not 9)

File: test/openjd/sessions/test_subprocess.py [hunk 4]
```diff
@@ -622,3 +633,222 @@ def end_proc():
...
+        # THEN
+        messages = collect_queue_messages(message_queue)
+        # If we printed "Trapped" then we hit our signal handler, and that shouldn't happen.
+        assert "Trapped" not in messages
+        # Check for the first message that would print
+        assert "Log from test 0" in messages
+        # If there's no 9, then we ended before the app naturally finished.
+        assert "Log from test 9" not in messages
+        assert subproc.exit_code != 0
+        assert len(children) == 2  # .sh & .py scripts parented under 'sudo'
```
Comment (matta-aws) [inline test/openjd/sessions/test_subprocess.py]: nit:  comment is posix oriented.  (no .sh script and no sudo on windows.)

File: test/openjd/sessions/test_subprocess.py (hunk 1 above)
Comment (roblutt) [reply test/openjd/sessions/test_subprocess.py]: Good point, changed

File: test/openjd/sessions/test_subprocess.py (hunk 3 above)
Comment (roblutt) [reply test/openjd/sessions/test_subprocess.py]: Fixed

File: test/openjd/sessions/test_subprocess.py (hunk 4 above)
Comment (roblutt) [reply test/openjd/sessions/test_subprocess.py]: Thanks, removed

**Review by matta-aws — APPROVED** [review]

**Review by ddneilson — APPROVED** [review]

**Pull request merged**